crates/bevy_transform/src/bundles.rs

```diff
@@ -17,25 +19,29 @@ use crate::prelude::{GlobalTransform, Transform};
 ///
 /// [`GlobalTransform`] is the position of an entity relative to the reference frame.
 ///
-/// [`GlobalTransform`] is updated from [`Transform`] by systems in the system set
+/// [`GlobalTransform`] is updated from [`Transform`] by systems in the system set. It is also
+/// automatically inserted whenever [`Transform`] is inserted. This is why this struct has
+/// been deprecated.
+///
 /// [`TransformPropagate`](crate::TransformSystem::TransformPropagate).
 ///
 /// This system runs during [`PostUpdate`](bevy_app::PostUpdate). If you
 /// update the [`Transform`] of an entity in this schedule or after, you will notice a 1 frame lag
 /// before the [`GlobalTransform`] is updated.
 #[derive(Clone, Copy, Debug, Default, Bundle)]
+#[deprecated(
+    since = "0.15.0",
+    note = "Use `Transform` on its own: `GlobalTransform` is now a required component."
+)]
 pub struct TransformBundle {
     /// The transform of the entity.
     pub local: Transform,
-    /// The global transform of the entity.
-    pub global: GlobalTransform,
```

Comment:
We shouldn't change this struct. We want to show the deprecation notice, not cause unnecessary compilation errors

In my opinion we can leave the docs as they were as well

Reply:
I'll revert the other bundles as well, meaning the migration will just be updated docs, deprecation, and removing other engine uses of TransformBundle/explicitly adding GlobalTransform.